client/src/globalState.tsx

```diff
@@ -164,13 +156,16 @@ export const GlobalStateProvider: React.FC<GlobalStateProviderProps> = ({
   const clearSelectedPinboard = () => setSelectedPinboardId(null);
 
   useEffect(() => {
-    if (isExpanded && !selectedPinboardId) {
-      preselectedPinboardQuery.refetch();
+    if (isExpanded && preselectedComposerId) {
+      preselectedPinboardQuery.stopPolling();
+      getPreselectedPinboard({
+        variables: { composerId: preselectedComposerId },
+      });
       preselectedPinboardQuery.startPolling(5000);
     } else {
       preselectedPinboardQuery.stopPolling();
     }
```

Comment:
`stopPolling` could be called in a returned cleanup function from `useEffect` - it might be a little cleaner (and also catch any edge case if global state were to somehow be unmounted and remounted)